### ISSUE TITLE
Respond to unhandledRejection event

### ISSUE THREAD

**Issue by arty-name** [getsentry/raven-node#144]
When using promises, thrown errors become rejected promises. It is easy to forget to handle a promise chain in some place. Would be nice to have something similar to `patchGlobal` for [`unhandledRejection`](https://nodejs.org/api/process.html#process_event_unhandledrejection). At least a mention of it in the configuration examples.

**Comment by benvinegar** [comment]
@arty-name – I'm still not convinced it should do this by default, but I will add a how-to to the examples, similar to [the ones up for raven-js](https://docs.getsentry.com/hosted/clients/javascript/usage/#promises).

**Comment by arty-name** [comment]
@benvinegar thank you! Even that would be nice!

**Comment by thomasboyt** [comment]
Am I right in assuming the correct way to do this would be:

```
process.on('unhandledRejection', (err) => {
  raven.captureException(err);
  // if you want to additionally exit on unhandled rejection:
  process.exit(1);
});
```

My only concern is that promise rejection reasons are often not `Error` instances, and I don't know how Raven/Sentry handle capturing "exceptions" that aren't actually exceptions.

**Comment by DamonGant** [comment]
You could always just wrap the string into an error object if you detect it.

I think the process event you mentioned is specific to bluebird, can anyone confirm?

**Comment by arty-name** [comment]
@DamonGant , this is an official node.js event, see the link in my first comment

**Comment by LewisJEllis** [comment]
Addressed by #219, closing.
